voc/python/klass.py

```diff
@@ -51,6 +58,10 @@ def __init__(
         self._constructor = None
 
     @property
+    def full_name(self):
+        return '.'.join(self.namespace.split('.')[1:] + [self.name])
```

Comment:
Good point—they do. Python uses `<module>.MyClass.InnerClass` in the `repr` result, so I guess we can follow this convention.

I wonder if this should be handled here, or in the class namespace. In the above example, should the inner classes’ namespaces be `<module>`, or `<module>.MyClass1` and `<module>.MyClass2`? And if it is the latter, how do we infer that inside the inner class (which does not have any information about the outer class, only the module it is in).

The answer is probably that we need to specify the namespace when creating the inner class in `add_class`:

```python
def add_class(self, ...):
    klass = Class(
        self.module,
        name=class_name,
        namespace=self.class_name,   # This.
        extends=extends,
        implements=implements,
    )
    ...
```

This would also affect both `descriptor` and `class_name` of the inner class. I’m not familiar enough with the code base to know if this would have unintended consequences.